frontend/src/actions/index.tsx

```diff
@@ -17,4 +17,30 @@ export interface DecrementAction extends Action {
   type: CounterTypes.DECREASE_COUNT
 }
 
-export type CounterActions = IncrementAction | DecrementAction;
\ No newline at end of file
+export type CounterActions = IncrementAction | DecrementAction
+
+export const SET_USER = 'SET_USER'
+export const LOG_OUT = 'LOG_OUT'
+
+export interface UserDetailsOnRegister {
+  email: string
+  username: string
+  password: string
+}
+
+export interface UserDetailsOnLogin {
+  email: string
+  password: string
+}
+
+export interface fetchUserAction {
+  type: typeof SET_USER
+  payload: UserDetailsOnRegister | UserDetailsOnLogin
+}
+
+export interface loggedoutUser {
+  type: typeof LOG_OUT
+  payload: null
```

Comment:
If we don't need payload here, then we don't need to define it. 
If we will be using generics I suggested above - you can use `void`, so it will be looks like:
`type LogoutUser = Action<AuthActionType.LOG_OUT>`